tests/vulture/run_vulture.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# Run vulture but ignore the following files:
+# - unit_tests: These contain many false positives because of the mock library
+EXCLUDE_PATTERNS="*unit_tests*"
+
+IGNORE_NAMES="""
+option_string # Used when creating custom actions for argparse
+"""
+# replace new line with commas and delete comments
+IGNORE_NAMES=$(echo "$IGNORE_NAMES" | tr '\n' ',' | sed 's/#.*//')
+
+find_scripts() {
+    # Helper to find all scripts in the tree
+    (
+        # Any non-binary file which contains a given shebang
+        git grep --cached -lIz '^#!.*'"$1"
+        shift
+        # Any file matching the provided globs
+        git ls-files -z "$@"
+    ) | sort -z | uniq -z
+}
+
+find_python_files() {
+    find_scripts 'python3' '*.py'
+}
```

Comment:
Will this autodetect files that dont have .py extension but are python executables?

Reply:
Good question. Based on description of the package

```
The provided arguments may be Python files or directories. For each directory Vulture analyzes all contained *.py files.
```
seems like not.

Reply:
In that case I would suggest to verify, and revert to the previous solution if my assumption is correct.

Reply:
From the docs I see ' For each directory Vulture analyzes all contained *.py files. ' 
So please revert to the previous script version, because that was finding also python executables which where not finishing in .py but had the shebang inside.

Reply:
@adamkankovsky please revert the script to the previous version, since this version skips all files that do not end in .py but are python executables.

Reply:
Agreed & IIRC we have quite a few such files.